### PR TITLE
feat(ShapeController): Add stop connections

### PR DESCRIPTION
**Pull request by KaylaBrady**
ticket: https://app.asana.com/0/1200180014510248/1203520428951853/f

This builds on https://github.com/mbta/skate/pull/1863 to surface route connections for all stops to comply with the data contract specified in https://www.notion.so/mbta-downtown-crossing/Stop-Connections-Implementation-b08dce1922704a95ad892af9e9535d18

### CODE REVIEW

File: lib/schedule/data.ex
Comment (lemald) [inline lib/schedule/data.ex:92]: **nitpick:** keeping the blank line between different functions is probably more readable.

**Comment by lemald** [comment]
@KaylaBrady this looks mostly good, though I'll hold off on final review + approval until #1863 is merged and this is rebased on to master. I'd be curious to see before and after comparisons of how long it takes to parse the schedule data with this change, of course it's going to add some overhead but I'm curious how much. Since the parsing process already takes several minutes on startup I want to make sure we don't make it too much slower.

File: lib/schedule/gtfs/stop.ex
Comment (firestack) [inline lib/schedule/gtfs/stop.ex:64]: praise: thanks for the doc comments!

comment: I liked the `\@spec` being at the top of the `\@doc` in the code like on [reject_connections_for_route](https://github.com/mbta/skate/blob/31f2a3041c8fc810b1e95cebd023476a79c7af4c/lib/schedule/gtfs/stop.ex#L51), but I don't have strong feelings about this one way or the other besides trying to decide what style we should use?

**Comment by KaylaBrady** [comment]
>  I'd be curious to see before and after comparisons of how long it takes to parse the schedule data with this change, of course it's going to add some overhead but I'm curious how much. Since the parsing process already takes several minutes on startup I want to make sure we don't make it too much slower.

@lemald I did some comparisons between this branch deployed to dev-blue, and a branch which reverts the parsing refactor entirely on dev-green. Results below! (Note there was a GTFS change since the logs compared in https://github.com/mbta/skate/pull/1863)

| Branch | Environment | memory | import duration (ms)  |   
|--------|-------------|--------|------------------|
|    kb-add-connections-data    |    skate-dev-blue         |     [link](https://mbta.splunkcloud.com/en-US/app/search/search?sid=1671573237.5616410)   |         [273544](https://mbta.splunkcloud.com/en-US/app/search/search?sid=1671573100.5615978)        |  
|  [kb-test-undo-refactor](https://github.com/mbta/skate/tree/kb-test-undo-refactor)      |       skate-dev-green      |    [link](https://mbta.splunkcloud.com/en-US/app/search/search?sid=1671573127.5616049)    |          [264258](https://mbta.splunkcloud.com/en-US/app/search/search?sid=1671572994.5615555)        |

**Review by lemald — APPROVED** [review]
The time difference in how long it takes to load the data is small enough that I think this is good to go. 👍

**Review by firestack — APPROVED** [review]

**Pull request merged**
